jose/jwk.py

```diff
@@ -218,46 +149,63 @@ class RSAKey(Key):
     This class requires PyCrypto package to be installed.
     This is based off of the implementation in PyJWT 0.3.2
     """
+
     SHA256 = Crypto.Hash.SHA256
     SHA384 = Crypto.Hash.SHA384
     SHA512 = Crypto.Hash.SHA512
+    valid_hash_algs = (SHA256, SHA384, SHA512)
 
-    def __init__(self, hash_alg):
-        self.hash_alg = hash_alg
+    prepared_key = None
+    hash_alg = None
 
-    def process_prepare_key(self, key):
+    def __init__(self, key, hash_alg):
+
+        if hash_alg not in self.valid_hash_algs:
+            raise JWKError('hash_alg: %s is not a valid hash algorithm' % hash_alg)
+        self.hash_alg = hash_alg
 
-        if isinstance(key, (_RSAKey, RSAKey)):
-            return key
+        if isinstance(key, _RSAKey):
+            self.prepared_key = key
+            return
 
         if isinstance(key, dict):
-            return self.process_jwk(key)
+            self.prepared_key = self.process_jwk(key)
+            return
 
         if isinstance(key, six.string_types):
             if isinstance(key, six.text_type):
                 key = key.encode('utf-8')
 
-            key = RSA.importKey(key)
-        else:
-            raise TypeError('Expecting a PEM- or RSA-formatted key.')
+            try:
+                self.prepared_key = RSA.importKey(key)
+            except Exception as e:
+                raise JWKError(e)
 
-        return key
+            return
 
-    def process_jwk(self, jwk):
+        raise JWKError('Unable to parse an RSA_JWK from key: %s' % key)
 
-        if not jwk.get('kty') == 'RSA':
-            raise JWKError("Incorrect key type.  Expected: 'RSA', Recieved: %s" % jwk.get('kty'))
+    def process_jwk(self, jwk_dict):
```

Comment:
is this method meant to be used directly? I don't see how somebody would want/need to call this on an instance of `RSAKey`, since you have to prepare a key to get an instance.

Making it `_process_jwk` would indicate that it is an implementation detail more than an API, if that's your intent